swap/simulations_test.go

```diff
@@ -0,0 +1,779 @@
+
+		// get the testService for the creditor
+		peerItem, ok := sim.NodeItem(creditor, bucketKeySwap)
+		if !ok {
+			return errors.New("no swap in simulation bucket")
+		}
+		creditorSvc := peerItem.(*testService)
+
+		for {
+			// let's always be nice and allow a time out to be catched
+			select {
+			case <-ctx.Done():
+				return errors.New("Timed out waiting for all swap peer connections to be established")
+			default:
+			}
+			// the node has all other peers in its peer list
+			if len(debitorSvc.swap.peers) == 1 && len(creditorSvc.swap.peers) == 1 {
+				break
+			}
+			// don't overheat the CPU...
+			time.Sleep(5 * time.Millisecond)
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
+		defer cancel()
+
+		// we will send just maxCheques number of cheques
+		maxCheques := 10
+
+		// the peer object used for sending
+		creditorPeer := debitorSvc.peers[creditor]
+
+		// send maxCheques number of cheques
+		for i := 0; i < maxCheques; i++ {
+			// use a price which will trigger a cheque each time
+			creditorPeer.Send(ctx, &testMsgBigPrice{})
+			// we need to wait a bit in order to give time for the cheque to be processed
+			err = waitForChequeProcessed(creditorSvc)
+			if err != nil {
+				return err
+			}
+		}
+
+		// check balances:
+		b1, err := debitorSvc.swap.loadBalance(creditor)
+		if err != nil {
+			return err
+		}
+		b2, err := creditorSvc.swap.loadBalance(debitor)
+		if err != nil {
+			return err
+		}
+
+		if b1 != -b2 {
+			return fmt.Errorf("Expected symmetric balances, but they are not: %d vs %d", b1, b2)
+		}
+		// check cheques
+		var cheque1, cheque2 *Cheque
+		if cheque1, err = debitorSvc.swap.loadLastSentCheque(creditor); err != nil {
+			return errors.New("expected cheques with creditor, but none found")
+		}
+		if cheque2, err = creditorSvc.swap.loadLastReceivedCheque(debitor); err != nil {
+			return errors.New("expected cheques with debitor, but none found")
+		}
+
+		// both cheques (at issuer and beneficiary) should have same cumulative value
+		if cheque1.CumulativePayout != cheque2.CumulativePayout {
+			return fmt.Errorf("Expected symmetric cheques payout, but they are not: %d vs %d", cheque1.CumulativePayout, cheque2.CumulativePayout)
+		}
+
+		// check also the actual expected amount
+		expectedPayout := uint64(maxCheques * (DefaultPaymentThreshold + 1))
+
+		if cheque2.CumulativePayout != expectedPayout {
+			return fmt.Errorf("Expected %d in cumulative payout, got %d", expectedPayout, cheque1.CumulativePayout)
+		}
+
+		return nil
+
+	})
+
+	if result.Error != nil {
+		t.Fatal(result.Error)
+	}
+
+	log.Info("Simulation ended")
+}
+
+// TestSimpleSimulation starts 16 nodes, then in a simple round robin fashion sends messages to each other.
+// Then checks that accounting is ok. It checks the actual amount of balances without any cheques sent,
+// in order to verify that the most basic accounting works.
+func TestSimpleSimulation(t *testing.T) {
```

Comment:
could you give a better name? maybe `TestSwapBasicSimulation`?

Reply:
Sure